packages/rpc/src/vstorageQuery.ts

```diff
@@ -0,0 +1,72 @@
+/* eslint-disable import/extensions */
+import axios from 'axios';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import axiosRetry from 'axios-retry';
+import type { FromCapData } from '@endo/marshal';
+import { AgoricChainStoragePathKind } from './types';
+
+export const pathToKey = (path: [AgoricChainStoragePathKind, string]) =>
+  path.join('.');
+
+export const keyToPath = (key: string) => {
+  const parts = key.split('.');
+  return [parts[0], parts.slice(1).join('.')] as [
+    AgoricChainStoragePathKind,
+    string,
+  ];
+};
+
+const parseIfJSON = (d: string | unknown) => {
+  if (typeof d !== 'string') return d;
+  try {
+    return JSON.parse(d);
+  } catch {
+    return d;
+  }
+};
+
+// Exported for testing.
+export const axiosClient = axios.create();
+
+axiosRetry(axiosClient, {
+  retries: 2,
+  retryDelay: axiosRetry.exponentialDelay,
+});
+
+export const vstorageQuery = async <T>(
+  node: string,
+  unmarshal: FromCapData<string>,
+  path: [AgoricChainStoragePathKind, string],
```

Comment:
consider naming these so they're clearer below
```suggestion
 [pathKind, path]: [AgoricChainStoragePathKind, string],
```

Reply:
Done